cloudsensei/lambda_function.py

```diff
@@ -0,0 +1,184 @@
+import json
+import os
+from time import time
+
+import boto3
+from datetime import datetime
+import requests
+
+
+class EC2Operations:
+    """
+    This class performs the ec2 operations
+    """
+
+    SLACK_ITEM_SIZE = 50
+
+    def __init__(self):
+        self.__ec2_client = boto3.client('ec2', region_name='us-east-1')
+        self.__iam_client = boto3.client('iam')
+
+    def set_ec2_client(self, region_name: str):
+        """
+        This method change the ec2_client object with another region
+        :param region_name:
+        :return:
+        """
+        self.__ec2_client = boto3.client('ec2', region_name=region_name)
+
+    def get_resources(self):
+        """
+        This method returns all the instances running more than 7 days
+        :return:
+        """
+        regions = self.__ec2_client.describe_regions()['Regions']
+        current_datetime = datetime.utcnow().date()
+        long_running_instances_by_user = {}
+        for region in regions:
+            region_name = region['RegionName']
+            self.set_ec2_client(region_name)
+            instances_list = self.__ec2_client.describe_instances()['Reservations']
+            for instances in instances_list:
+                for resource in instances['Instances']:
+                    launch_time = resource.get('LaunchTime').date()
+                    days = (current_datetime - launch_time).days
+                    if days > 7:
```

Comment:
Why don't we make this configurable instead of hardcoding to 7?

Reply:
@smalleni we can make it configurable, will default it to 7 days.